tests/test_dupefilters.py

```diff
@@ -129,6 +134,26 @@ def request_fingerprint(self, request):
 
         case_insensitive_dupefilter.close('finished')
 
+    def test_seenreq_newlines(self):
+        """ Checks against adding duplicate \r to
+        line endings on Windows platforms. """
+
+        r1 = Request('http://scrapytest.org/1')
+
+        path = tempfile.mkdtemp()
+        try:
+            df = RFPDupeFilter(path)
+            df.open()
+            df.request_seen(r1)
+            df.close('finished')
+
+            with open(os.path.join(path, 'requests.seen'), 'rb') as seen_file:
+                line = next(seen_file).decode()
+                assert not line.endswith('\r\r\n')
```

Comment:
Could you also check if the line ends with `\n` here?

Reply:
Just updated the PR, how's that? It should check that universal line endings mode is selected and working as expected now on each platform (Linux/Mac and Windows).